src/ngx_http_lua_tcpcong.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) Rain Li (blacktear23)
+ *
+ */
+
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+
+#include "ngx_http_lua_tcpcong.h"
+#include "ngx_http_lua_util.h"
+#include "ngx_http_lua_api.h"
+
+
+int ngx_socket_set_tcp_congestion(ngx_socket_t s, const char* cong_name);
+static int ngx_http_lua_ngx_set_tcp_congestion(lua_State *L);
+
+
+#if (NGX_LINUX)
+
+int
+ngx_socket_set_tcp_congestion(ngx_socket_t s, const char* cong_name) {
+    u_char optval[16];
+    ngx_cpystrn(optval, (u_char *)cong_name, 16);
+    return setsockopt(s, IPPROTO_TCP, TCP_CONGESTION,
+                      (void *)optval, strlen((char *)optval));
+}
+
+#else
+
+int
+ngx_socket_set_tcp_congestion(ngx_socket_t s, const char* cong_name) {
```

Comment:
Style: `{` should be in the separate line according to the nginx coding style. BTW, you can always find such things via the `ngx-releng` tool here:

https://github.com/openresty/openresty-devel-utils/blob/master/ngx-releng

Reply:
Also, we should use a space *before* instead of after `*` in the pointer type signature. See other similar places for examples.